nuntium/views.py

```diff
@@ -92,14 +96,21 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class MessageDetailView(DetailView):
-    template_name = 'nuntium/message/message_detail.html'
+class MessageDetailViewMixin(DetailView):
     model = Message
 
     def get_queryset(self):
```

Comment:
I think you could override `get_object` here rather than `get_queryset`. What you really want is to find the unique Message object which has slug `self.kwargs['slug']`, isn't it? I'm currently confused as to how PostSubmissionView finds its object.